photon/src/http.rs

```diff
@@ -56,9 +56,12 @@ fn bake_ctx(inp: &str, ctx: &Context) -> Option<String> {
 
         let mut updated = 0;
         for mat in matches.iter() {
-            let compiled = compile_expression(&mat.as_str()[2..mat.len() - 2]);
+            let compiled = compile_expression_validated(
+                &mat.as_str()[2..mat.len() - 2],
```

Comment:
Possible small improvement: this depends on the opening/closing bracket always being exactly two bytes long. Maybe we could specify the regex like `r"\{\{([^{}]*)}}"` and use capture groups to extract the innards?